web/src/stream_ui_updates.js

```diff
@@ -324,15 +335,20 @@ export function update_stream_row_in_settings_tab(sub) {
     // This is in the left panel.
     // This function display/hide stream row in stream settings tab,
     // used to display immediate effect of add/removal subscription event.
-    // If user is subscribed to stream, it will show sub row under
-    // "Subscribed" tab, otherwise if stream is not public hide
-    // stream row under tab.
-    if (is_subscribed_stream_tab_active()) {
-        const $sub_row = row_for_stream_id(sub.stream_id);
-        if (sub.subscribed) {
-            $sub_row.removeClass("notdisplayed");
+    // If user is subscribed or unsubscribed to stream, it will show sub or unsub
+    // row under "Subscribed" or "Not subscribed" (only if the stream is public) tab, otherwise
+    // if stream is not public hide stream row under tab.
+
+    if (is_subscribed_stream_tab_active() || is_not_subscribed_stream_tab_active()) {
+        const $row = row_for_stream_id(sub.stream_id);
+
+        if (
+            (is_subscribed_stream_tab_active() && sub.subscribed) ||
+            (is_not_subscribed_stream_tab_active() && !sub.subscribed)
+        ) {
+            $row.removeClass("notdisplayed");
         } else if (sub.invite_only || current_user.is_guest) {
```

Comment:
This becomes confusing as we show the non-subscribed stream in the tab even when the stream is private, but then the stream is hidden ultimately as the client will receive the "remove" event. So, the behavior is fine but we can probably improve the code to avoid any future bugs.

Not sure what would be the best way to handle this, but just mentioning it here.

Reply:
Currently we have same behavior if private stream is unsubscribed it is shown in the all streams tab.

Reply:
I think it is shown for admins only and not non-admins.

Just to be clear, the behavior in the PR is correct, I am just not sure whether we should remove the stream here only or we should wait for "remove" event.

Reply:
should I ask it on CZO ?

Reply:
I think we can wait for a review by Tim first.

Reply:
I think this is probably fine for now but we should think about it as we work on the archived streams transition.

Reply:
Since it was related to how to implement it better considering that code is not confusing to read and not how the app would behave, I thought would be better to just discuss it here where the code changes can be accessed easily. Will take care next time.